src/object/interrupt.c

```diff
@@ -182,74 +182,81 @@ void deletedIRQHandler(irq_t irq)
 void handleInterrupt(irq_t irq)
 {
     if (unlikely(IRQT_TO_IRQ(irq) > maxIRQ)) {
-        /* mask, ack and pretend it didn't happen. We assume that because
-         * the interrupt controller for the platform returned this IRQ that
-         * it is safe to use in mask and ack operations, even though it is
-         * above the claimed maxIRQ. i.e. we're assuming maxIRQ is wrong */
-        printf("Received IRQ %d, which is above the platforms maxIRQ of %d\n", (int)IRQT_TO_IRQ(irq), (int)maxIRQ);
+        /* The interrupt number is out of range. Pretend it did not happen by
+         * handling it like an inactive interrupt (mask and ack). We assume this
+         * is acceptable, because the platform specific interrupt controller
+         * driver reported this interrupt. Maybe the value maxIRQ is just wrong
+         * it set to a lower value because the interrupts are unused.
+         */
+        printf("Received IRQ %d, which is above the platforms maxIRQ of %d\n",
+               (int)IRQT_TO_IRQ(irq), (int)maxIRQ);
         maskInterrupt(true, irq);
-        ackInterrupt(irq);
-        return;
-    }
-    switch (intStateIRQTable[IRQT_TO_IDX(irq)]) {
```

Comment:
The GitHub diff is hard to read at this point. It looks like this only factors out `ackInterrupt` and moves it down, but I don't think we should change the structure here. This looks like a small change, but the early return makes the proof nicer in this case and we'd lose that with the big `else`. I'd rather have the small duplication in the code than touch those proofs (they are pretty hairy).

If we remove `Arch_finaliseInterrupt`, which now does seem possible, this function could actually remain unchanged.

Reply:
yes. I have move this to the end of the function to avoid code duplication because `ackInterrupt()` is done on any case and `Arch_finaliseInterrupt()` also.